libp2p/builders.nim

```diff
@@ -109,22 +125,28 @@ proc withAgentVersion*(b: SwitchBuilder, agentVersion: string): SwitchBuilder =
 
 proc build*(b: SwitchBuilder): Switch =
   if b.rng == nil: # newRng could fail
-    raise (ref CatchableError)(msg: "Cannot initialize RNG")
+    raise newException(Defect, "Cannot initialize RNG")
+
+  let pkRes = PrivateKey.random(b.rng[])
+  if pkRes.isErr:
+    raise newException(Defect, "No RNG supplied!")
 
   let
-    seckey = b.privKey.get(otherwise = PrivateKey.random(b.rng[]).tryGet())
+    seckey = b.privKey.get(otherwise = pkRes.get())
```

Comment:
generally, it's better to use `.expect("reason why it's safe")` - `get` panics with a poor error message, but above all, when reading the logic, it helps make clear why it's safe to assume that there's a value stored in here